imgui.cpp

```diff
@@ -9158,6 +9161,31 @@ const ImGuiPayload* ImGui::GetDragDropPayload()
     return g.DragDropActive ? &g.DragDropPayload : NULL;
 }
 
+const ImGuiPayload* ImGui::AcceptReorderDropPayload(const char* type, ImGuiDragDropFlags flags)
+{
+    ImGuiContext& g = *GImGui;
+    const ImGuiPayload* payload = NULL;
+    ImRect bb(GetCursorScreenPos(), GetCursorScreenPos());
+
+    if (flags & ImGuiDragDropFlags_ReorderHorizontal)
+    {
+        bb.Min.x -= g.Style.ItemSpacing.x;
+        bb.Max.y += g.FontSize;
+    }
+    else
+    {
+        bb.Max.x += GetContentRegionAvail().x;
+        bb.Min.y -= g.Style.ItemSpacing.y;
+    }
+
```

Comment:
Not sure I understand the logic there, why is `g.FontSize` used? Is that assuming a certain size for items?
For trees with indentation we'll also want way for the drop target so cover whole width (like `ImGuiTreeNodeFlags_SpanFullWidth`) so maybe we need more use of this api before designing the thing.

Reply:
> why is `g.FontSize` used

I intended to get a line height here. Is there a better way? Admittedly font sizes are bit confusing to me, lots of different variables that affect font size.

Reply:
Why a "line height" for a separator between item? Shouldn't the separator be very thin (roughly ~style.ItemSpacing.y?).

Text-only line are `g.FontSize` high, most lines with framed widgets are `g.FontSize + style.FramePadding.y * 2`

Reply:
Ah I think you want to use `window->DC.CurrLineSize.y` here?